qualtran/qref_interop/bartiq_demo.ipynb

```diff
@@ -0,0 +1,2214 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "bafd2405-3ac0-43ff-b9b4-7978ffd49cef",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "TComplexity(t=20, clifford=106, rotations=2)\n",
+      "[ResourceV1(name='clifford', type='additive', value=106),\n",
+      " ResourceV1(name='rotations', type='additive', value=2),\n",
+      " ResourceV1(name='t', type='additive', value=20)]\n"
+     ]
+    }
+   ],
+   "source": [
+    "USP_bloq = [bloq_inst.bloq for bloq_inst in bloq.bloq_instances if bloq_inst.bloq.pretty_name() == \"PrepareUniformSuperposition\"][0]\n",
+    "pprint(USP_bloq.t_complexity())\n",
+    "pprint(qref_definition.program.children.by_name[\"PrepareUniformSuperposition_0\"].resources)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4953f807-5ce0-42c5-9bf4-6773c5f45e87",
+   "metadata": {},
+   "source": [
+    "Now we can take the QREF definition and pass it to Bartiq.\n",
+    "However, it would be trivial as all the resources are already there. For this reason, before passing it to Bartiq, we'll remove the top level resources in QREF."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "56beb66e-0989-4cf6-920b-bbc51cae559d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'clifford': <Resource name=\"clifford\" value=\"1770\">,\n",
+      " 'rotations': <Resource name=\"rotations\" value=\"2\">,\n",
+      " 't': <Resource name=\"t\" value=\"307\">}\n"
+     ]
+    }
+   ],
+   "source": [
+    "from bartiq.integrations.qref import qref_to_bartiq\n",
+    "from bartiq import compile_routine\n",
+    "\n",
+    "qref_definition.program.resources =  [] \n",
+    "bartiq_routine = qref_to_bartiq(qref_definition)\n",
+    "compiled_routine = compile_routine(bartiq_routine)\n",
+    "pprint(compiled_routine.resources)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2db6c501-5756-4356-ae6f-a6f412155e68",
+   "metadata": {},
+   "source": [
+    "As we can see, Bartiq correctly reconstructed the costs from the subroutines."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "27483d83-d008-499c-9322-c2953f6db573",
+   "metadata": {},
+   "source": [
+    "## Writing custom USP implementation\n",
+    "\n",
+    "So far so good, but we used Bartiq to derive the same information we already had in Qualtran.\n",
+    "\n",
+    "There are two limitations of Qualtran implementation, with which Bartiq can help us, let's see what these are.\n",
+    "\n",
+    "### Numeric implementation\n",
+    "\n",
+    "The example above is purely numeric. We can get the symbolic cost of the whole `StatePreparationAliasSampling` (see snippet below) but we can't decompose it using `decompose_bloq()`.\n",
+    "\n",
```

Comment:
This will be fixed once https://github.com/quantumlib/Qualtran/pull/1084 is merged; so symbolic decomposition would work as expected for the given bloq example

Reply:
Ok, will fix it then :)